docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,340 @@ To make the actions more user-friendly you can also use alternating buttons as
+documentation available all over the web. We recommend:
+* [The Official React Tutorial](https://facebook.github.io/react/tutorial/tutorial.html)
+* [Build With React](http://buildwithreact.com/tutorial)
+
+#### A few words about ES6
+The remainder of this tutorial is written in [ECMAScript 6](http://es6-features.org/#Constants), or _ES6_
+for short. This is the new spec for Javascript (currently ES5) that is as of this writing
+only partially implmented in modern browsers. Because it doesn't yet enjoy vast native support,
+it has to be [transpiled](https://www.stevefenton.co.uk/2012/11/compiling-vs-transpiling/) in order to work
+in a browser. This transpiling can be done using a variety of toolchains, but the basic
+ principle is that a browser-ready, ES5 version of your code is generated in your dev
+ environment as part of your workflow. Often called a "bundle," you should rarely have 
+ to see this file. It is effectively an invisible layer that translates modern ES6
+ code into something a browser can parse. As browsers evolve, this step will become less
+ necessary. (Although, it is worth noting that because transpiling comes at such a low cost,
+  and browsers are relatively slow to catch up, we'll probably be using it for the 
+  foreseeable future in order to adopt new features beyond ES6).
+   
+   As stated above, there are many ways to solve the problem of transpiling. The toolchain
+   we use in core SilverStripe modules includes:
+   * [Babel](http://babeljs.io) (ES6 transpiler)
+   * [Webpack](http://webpack.js.org) (Module bundler)
+ 
+### Customising React components
+
+React components can be customised in a similar way to PHP classes, using a dependency
+injection API. The key difference is that components are not overriden the way backend
+services are. Rather, new components are composed using [higher order components](https://facebook.github.io/react/docs/higher-order-components.html).
+This has the inherent advantage of allowing all thidparty code to have an influence
+over the behaviour, state, and UI of a component.
+
+#### A simple higher order component
+
+Using our example above, let's create a customised `PhotoItem` that allows a badge,
+perhaps indicating that it is new to the gallery.
+
+```js
+const enhancePhoto = (PhotoItem) => (props) {
+	const badge = props.isNew ? 
+	  <div className="badge">New!</div> : 
+	  null;
+
+	return (
+		<div>
+			{badge}
+			<PhotoItem {...props} />
+		</div>
+	);
+}
 
-### Creating a component
+const EnhancedPhotoItem = enhancedPhoto(PhotoItem);
 
-__my-component.js__
-```javascript
-import SilverStripeComponent from 'silverstripe-component';
+<EnhancedPhotoItem isNew={true} size={300} />
+```
 
-class MyComponent extends SilverStripeComponent {
+Alternatively, this component could be expressed with an ES6 class, rather than a simple
+function.
+
+```js
+const enhancePhoto = (PhotoItem) => {
+	return class EnhancedPhotoItem extends React.Component {
```

Comment:
Oooh, looks like you've done something smart with `createDisplayName` there. Using the module name as a fallback kind of works, but we shouldn't encourage skipping `displayName`. `my-other-module(my-module(TextField)))` isn't as expressive as `TextLengthChecker(CharacterCounter(TextField)))`